libr/include/rvc.h

```diff
@@ -0,0 +1,47 @@
+/* radare - LGPL - Copyright 2013-2020 - RHL120, pancake */
+
+#include <r_core.h>
+#include <r_util.h>
+
+typedef struct blob {
+	char *fname;
+	char *hash;
+} RvcBlob;
+
+typedef struct commit {
+	struct commit *prev;
+	RList *blobs;
+	char *author;
+	int64_t timestamp;
+	char *hash;
+	size_t next_num;
+	bool ishead;
+	char *message;
+	struct commit *next;
+} RvcCommit;
+
+typedef struct branch {
+	char *name;
+	RvcCommit *head;
+} RvcBranch;
+
+typedef struct RVc {
+	char *path;
+	RList *branches;
+	RvcBranch *current_branch;
+} Rvc;
+
+
+
+R_API RvcBlob *rvc_path_to_commit(Rvc *repo, const char *path);
+R_API RList *rvc_uncomitted(Rvc *repo);
+R_API bool rvc_checkout(Rvc *repo, const char *name);
+R_API bool rvc_commit(Rvc *repo, RList *blobs, const char *auth, const char *message);
+R_API bool rvc_branch(Rvc *repo, const char *name);
+R_API RList *rvc_add(Rvc *repo, RList *files);
+R_API Rvc *rvc_new(const char *path);
+R_API int git_init (const char *path);
```

Comment:
better rename all those git_ functions to `r_vc_git` instead, and the `rvc_` ones to `r_vc_*` this way it wont conflict with libgit , the `r_` is sacred :D

Reply:
I kind of wanted to have a clear seperation between the rvc apis & the git apis but your idea seems simpler & more flexible